tests/testthat/test-secondary_weights.R

```diff
@@ -0,0 +1,99 @@
+# Test secondary_weights()
+# test_that("secondary_weights matches key", {
+#
+#   # Run code
+#   secondary_weights_cropped_output <- secondary_weights(
+#     secondary_raster = cropland_nj_2015,
+#     grid = era5_grid,
+#     extent = c(-75.75, -73.5, 38.75, 41.25)
+#   ) %>%
+#     dplyr::arrange(x, y)
+#
+#   # Load key generate by previous run
+#   secondary_weights_cropped_key <- readRDS(
+#     testthat::test_path("fixtures/secondary_weights_cropped_key.rds")
+#   ) %>%
+#     dplyr::arrange(x, y)
+#
+#   # Expect identical
+#   expect_true(all.equal(
+#     secondary_weights_cropped_output,
+#     secondary_weights_cropped_key
+#   ))
+#
+# })
+
+
+# Test secondary_weights() without checking actual values in output
+test_that("secondary_weights outputs are normal", {
+
+  # Run secondary_weights
+  output <- secondary_weights(cropland_nj_2015)
+
+  # Expect the correct column names
+  expect_equal(names(output), c("x", "y", "weight"))
+
+  # Expect the correct number of rows
+  expect_equal(nrow(output), 182)
+
+  # Expect that area = length * width
+  expect_equal(length(unique(output$x)) * length(unique(output$y)),
+               182)
+
+  # Expect no NAs with normal output
+  expect_true(all(!is.na(output$weight)))
+
+})
+
+## Currently this test isn't passing
+# test_that("secondary_weights handles NAs",{
+#
+#   ## Add NAs
+#   crop_na <- raster::reclassify(cropland_nj_2015, cbind(0,0.9,NA))
+#   options(warn=-1)  # This won't actually run within the testing code because of the warning so we suppress it
+#   output_na <- secondary_weights(crop_na, era5_grid, "full")
+#
+#   na_rows <- output_na %>%
+#     dplyr::filter(is.na(weight))
+#
+#   # Expect that output_na does include NA values
+#   expect_true(nrow(na_rows) > 0)
+# })
+
```

Comment:
@tliddell4 maybe we can remove this one too for now and keep the code somewhere else until we can fix it? I feel like we shouldn't have tests in the main branch that we aren't using or don't work